components/org.wso2.micro.integrator.security/src/main/java/org/wso2/micro/integrator/security/vault/VaultTool.java

```diff
@@ -195,4 +217,24 @@ private static String doEncryption(Cipher cipher, String plainTextPwd) {
         System.out.println("\nEncryption is done Successfully\n");
         return encodedValue;
     }
+
+    private static void addSystemProperties() {
+        Properties properties = new Properties();
+        String filePath = System.getProperty(CONF_DIRECTORY_PATH) + File.separator + CARBON_PROPERTIES;
+        File file = new File(filePath);
+
+        if (file.exists()) {
+            try ( InputStream inputStream = new FileInputStream(file)) {
+                properties.load(inputStream);
+            } catch (IOException e) {
+                e.printStackTrace();
+                System.exit(1);
+            }
+        }
+
+        String secureVaultRegex = properties.getProperty(Constants.SECURE_VALULT_PASSWORD_REGEX_SYSTEM_PROPERTY);
+        if (secureVaultRegex != null) {
```

Comment:
Do we need to have a null check here?

Reply:
Changed

Reply:
Null check is required here as setting a property with a null value throws a NullPointerException. 

Related PR : https://github.com/wso2/micro-integrator/pull/841